pkg/component/shared/kubeapiserver.go

```diff
@@ -80,7 +81,7 @@ func init() {
 func NewKubeAPIServer(
 	ctx context.Context,
 	runtimeClientSet kubernetes.Interface,
-	auditConfigClient client.Client,
+	gardenClient client.Client,
```

Comment:
```suggestion
	resourceConfigClient client.Client,
```

As it's not the `gardenClient` in the Operator case ([ref](https://github.com/gardener/gardener/blob/498403ac7be3f528d51e3647882155414f22289e/pkg/operator/controller/garden/components.go#L442))